pennylane/wires.py

```diff
@@ -384,15 +401,17 @@ def all_wires(list_of_wires, sort=False):
         >>> Wires.all_wires(list_of_wires)
         <Wires = [4, 0, 1, 3, 5]>
         """
-
         combined = []
+        seen_labels = set()
         for wires in list_of_wires:
             if not isinstance(wires, Wires):
                 raise WireError(
                     "Expected a Wires object; got {} of type {}".format(wires, type(wires))
                 )
 
-            combined.extend(wire for wire in wires.labels if wire not in combined)
+            extension = [label for label in wires.labels if label not in seen_labels]
+            combined.extend(extension)
+            seen_labels.update(extension)
```

Comment:
So in the previous code, The code `if wire not in combined` would loop through all of `combined` for every `wire in wires.labels`, resulting in a O(n^2) runtime. By having `seen_labels` be a set, the `if label not in seen_labels` is only O(1) time, bringing the total runtime to O(n).

Reply:
> And why not just turn extension into a set, instead of updating seen_labels?

`combined` requires the order to remain the same, so `combined.extend(extension)` would have an indeterministic order if `extension` was a set.